st2common/st2common/util/schema/__init__.py

```diff
@@ -146,7 +149,9 @@ def assign_default_values(instance, schema):
                                                                 schema=array_schema)
 
         # Object
-        if attribute_type == 'object' and property_data.get('properties', {}):
+        if (property_data.get('properties', {}) and
```

Comment:
Imo, it would make it easier to read it we store those in temporary variable and do something like:

```python
if has_properties and (is_object or is_list_and_has_object):
```